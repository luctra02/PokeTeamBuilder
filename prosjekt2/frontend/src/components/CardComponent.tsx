import ChangeTeamComponent from './ChangeTeamComponent';
import TypeComponent from './TypeComponent';

interface PokemonObjectType {
  id: number;
  name: string;
  image: string;
  types: string[];
  key: string;
  weight: number;
  height: number;
  baseStats: number[];
}

interface UpdateCountFunction {
  (newCount: number): void;
}

function CardComponent({
  pokemonObject,
  updateCount,
}: {
  pokemonObject: PokemonObjectType;
  updateCount?: UpdateCountFunction;
}) {
  return (
    <div className="pokemonCard">
      <h2>
        #{pokemonObject.id} {pokemonObject.name}
      </h2>
      <img src={pokemonObject.image} alt="" />
      <div className="pokeType">
        {pokemonObject.types.map((type, index) => (
          <TypeComponent key={index} pokemonType={type} />
        ))}
      </div>
      <ChangeTeamComponent
        updateCount={updateCount}
        pokemonTeam={{
          num: pokemonObject.id,
          sprite: pokemonObject.image,
          types: pokemonObject.types,
          name: pokemonObject.name,
          baseStats: pokemonObject.baseStats,
          key: pokemonObject.key,
          weight: pokemonObject.weight,
          height: pokemonObject.height,
        }}
      />
    </div>
  );
}

export default CardComponent;
