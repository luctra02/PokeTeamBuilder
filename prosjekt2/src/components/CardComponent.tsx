import TypeComponent from "./TypeComponent";

interface pokemonObjectType {
  id: string;
  name: string;
  sprites: string
  types: {
    type: {
      name: string;
    };
  }[];

}

function CardComponent(pokemonObject: pokemonObjectType) {
  return (
    <div className="pokemonCard">
      <h1>{pokemonObject.name}</h1>
      <img src={pokemonObject.sprites} alt="" />
      <div className="pokeType">
        {pokemonObject.types.map((type) => (
          <TypeComponent pokemonType={type.type.name} />
        ))}
      </div>
    </div>
  );
}

export default CardComponent