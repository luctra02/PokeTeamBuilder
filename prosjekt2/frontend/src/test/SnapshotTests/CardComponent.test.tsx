import { screen, waitFor } from '@testing-library/react';
import { getPokemonsMock, allMocks as mocks} from '../mocks/mockQueries';
import { testPageRender } from '../testUtils';
import CardComponent from '../../components/CardComponent';




describe('renders a pokemon', () => {
    it('matches the snapshot', async () => {
      const { asFragment } = testPageRender(<CardComponent pokemonObject= {getPokemonsMock.result.data.getPokemons.pokemons[0]}/>, { mocks });
  
       await waitFor(() => {
        expect(screen.getByText(/Pikachu/i)).toBeInTheDocument();
        expect(screen.getByText(/#0025/i)).toBeInTheDocument();
        expect(screen.getByText(/Electric/i)).toBeInTheDocument();
      });
  
      expect(asFragment()).toMatchSnapshot();
    });
  }); 
