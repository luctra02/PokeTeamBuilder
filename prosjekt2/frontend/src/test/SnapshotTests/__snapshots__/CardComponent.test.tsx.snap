// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders a pokemon > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="pokemonCard"
  >
    <h2>
      Pikachu
    </h2>
    <img
      alt=""
      src="https://play.pokemonshowdown.com/sprites/ani/pikachu.gif"
    />
    <div
      class="pokemonID"
    >
      #0025
    </div>
    <div
      class="pokeType"
    >
      <div>
        <div
          class="pokeType PokeTypeBox"
          style="background-color: rgb(248, 208, 48);"
        >
          Electric
        </div>
      </div>
    </div>
    <button
      class="teamButton"
    >
      Add to Team
    </button>
  </div>
</DocumentFragment>
`;
